File: backend/api/resume.py
from fastapi import APIRouter, UploadFile, File, HTTPException
from fastapi.responses import JSONResponse
import os
import tempfile
from typing import Optional

from agent.resume_agent import ResumeAgent

router = APIRouter()
resume_agent = ResumeAgent()

@router.post("/resume/upload")
async def upload_resume(file: UploadFile = File(...)):
    """
    上传简历文件（PDF/DOCX）并提取结构化信息
    """
    # 验证文件类型
    allowed_extensions = {'.pdf', '.docx'}
    file_extension = os.path.splitext(file.filename)[1].lower()
    
    if file_extension not in allowed_extensions:
        raise HTTPException(
            status_code=400,
            detail=f"不支持的文件格式。仅支持: {', '.join(allowed_extensions)}"
        )
    
    # 保存临时文件
    try:
        with tempfile.NamedTemporaryFile(delete=False, suffix=file_extension) as tmp_file:
            content = await file.read()
            tmp_file.write(content)
            tmp_file_path = tmp_file.name
        
        # 调用 ResumeAgent 处理
        result = resume_agent.parse_resume(tmp_file_path, file.filename)
        
        # 清理临时文件
        os.unlink(tmp_file_path)
        
        return JSONResponse(
            status_code=200,
            content={
                "success": True,
                "resume_id": result.get("resume_id"),
                "data": result.get("data")
            }
        )
    
    except Exception as e:
        # 清理临时文件
        if 'tmp_file_path' in locals() and os.path.exists(tmp_file_path):
            os.unlink(tmp_file_path)
        
        raise HTTPException(
            status_code=500,
            detail=f"简历处理失败: {str(e)}"
        )

File: frontend/src/api/resume.js
import request from '../utils/request'

/**
 * 上传简历文件（PDF/DOCX），提取结构化信息
 * @param {File} file - 简历文件
 * @returns {Promise<{success: boolean, resume_id: string, data: object}>}
 */
export async function uploadResume(file) {
  const formData = new FormData()
  formData.append('file', file)
  return request.post('/resume/upload', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

/**
 * 解析岗位 JD 文本
 * @param {string} jobText - JD 文本内容
 * @returns {Promise<{success: boolean, data: object}>}
 */
export async function parseJd(jobText) {
  return request.post('/parse_jd', { job_text: jobText })
}